go.mod

```diff
@@ -5,47 +5,216 @@ go 1.23.0
 toolchain go1.23.4
 
 require (
-	cloud.google.com/go v0.110.2
+	cloud.google.com/go v0.116.0
 	github.com/LastPossum/kamino v0.0.2
-	github.com/Masterminds/semver/v3 v3.2.1
+	github.com/Masterminds/semver/v3 v3.3.0
 	github.com/asaskevich/govalidator v0.0.0-20230301143203-a9d515a09cc2
 	github.com/go-jose/go-jose/v4 v4.0.5
-	github.com/google/go-cmp v0.6.0
+	github.com/golangci/golangci-lint v1.64.8
```

Comment:
This seems to add a massive amount of sub-dependencies, I'm not convinced... is there a link to materials where this is suggested as best practice?

Reply:
TIL since Go 1.24 the [`tools` directive](https://go.dev/doc/go1.24#tools) is available for `go.mod`. If you decide to upgrade, then that would be the official way. As you can see, there they mention the use of `tools.go` as the previous workaround. For reference, [here's the comment I mentioned in our chat](https://github.com/golang/go/issues/25922#issuecomment-413898264) where it was regarded as the best approach back in 2018.

In any case, since Go 1.17 it seems that [any modules not needed by the `main` package are automatically pruned out](https://go.dev/ref/mod#graph-pruning) (if I understood correctly) so either way it _shouldn't_ affect build or test times.

LMK what your preference is, I'll adapt the PR accordingly!

Reply:
Okay, thanks for that, makes more sense now. I think what's lacking however is an example of how this is actually used. We have examples for Go generate, but its not clear how to use the tools.

We can upgrade to 1.24 for this public package as its not nice to force upgrades. Its already on 1.23 due to an external dependency.

Reply:
> We can upgrade to 1.24 for this public package as its not nice to force upgrades. Its already on 1.23 due to an external dependency.

It's not clear to me if you are saying we _can_ or _can't_ upgrade. I assume it's the latter, but since Go versions are backwards-compatible, [upgrading minor versions should never break](https://tip.golang.org/doc/go1compat).

And as for the example I'll do it tomorrow after work, should be fairly easy!